WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/StartingInfo/PromptProgressBar/ProductCreationAIPromptProgressBarViewModel.swift

```diff
@@ -0,0 +1,133 @@
+import SwiftUI
+
+class ProductCreationAIPromptProgressBarViewModel: ObservableObject {
+    @Published private(set) var status: ProgressStatus = .start
+    @Published var text: String = "" {
+        didSet {
+            updateStatusBasedOnText(text)
+        }
+    }
+
+    // Method for updating the text
+    func updateText(to newText: String) {
+        text = newText
+    }
+
+    // Method for updating the status based on text
+    private func updateStatusBasedOnText(_ text: String) {
+        let wordCount = text.split { $0 == " " || $0.isNewline }.count
+
+        switch wordCount {
+        case 0:
+            status = .start
+        case 1...10:
+            status = .inProgress
+        case 11...19:
+            status = .halfway
+        case 20...30:
+            status = .almostDone
+        default:
+            status = .completed
```

Comment:
Opinion: Based on my testing, the `wordCount` we defined here looks like a strict expectation. Do you think we can reduce the expected word count? If it makes sense, consider starting a slack thread to discuss this further with the team. 

<img src="https://github.com/woocommerce/woocommerce-ios/assets/524475/737dc074-77b1-4942-810b-57a65bf8fe93" width="320"/>

Reply:
I don’t have a strong opinion tbh. I think is tricky to come up with good numbers that will work well for all languages.  But yeah, I think overall we can maybe reduce all the ranges by 2-3 words?

Reply:
Discussion moved here C03L1NF1EA3/p1720706347936069

Reply:
Merging this now, I can update the ranges later on based on the results of the discussion.